### PR TITLE
AGD-1437  Can not preload a package with an apostrophe in its the folder path.

### PR DESCRIPTION
**Pull request by saintentropy**
### Purpose

This PR is to resolve an issue found with localized French samples for GD where the included packages would not load into Dynamo.  We found while these packages will load during runtime (ie if the user adds a folder to the "Manage Node and Package Paths" dialog) the package will not load upon a restart of Dynamo.  This PR adds a fix for the formatted string path which includes an apostrophe used to preload packages during start up.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@QilongTang @mrahmaniasl 

### FYIs

@mjkkirschner

### CODE REVIEW

File: src/Engine/ProtoCore/Utils/FileUtils.cs [hunk 1]
```diff
@@ -23,6 +23,9 @@ public static string GetDSFullPathName(string fileName, Options options = null)
             var trimChars = new[] {'\n','\t','\r',' '};
             fileName = fileName.Trim(trimChars);
 
+            // Fix file paths which include an apostrophe
+            fileName = fileName.Replace("\\'", "'");
```
Comment (QilongTang) [inline src/Engine/ProtoCore/Utils/FileUtils.cs:27]: If I understand it correctly this is replacing `\'` with `'`, but how would the test case reflect this fix? Maybe I missed something?

File: test/DynamoCoreTests/DSLibraryTest.cs [hunk 2]
```diff
@@ -121,6 +121,32 @@ public void TestLoadDllFileSuccess()
...
+            Assert.IsTrue(LibraryLoaded);
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void TestLoadDllFileWithApostropheInPathSuccess()
+        {
+            LibraryLoaded = false;
+
+            string libraryPath = Path.Combine(TestDirectory, "pkgs", "Pack'age", "bin", @"Package.dll");
```
Comment (saintentropy) [inline test/DynamoCoreTests/DSLibraryTest.cs:143]: @QilongTang This test fails with out the formatting fix

Comment (QilongTang) [reply test/DynamoCoreTests/DSLibraryTest.cs:143]: @saintentropy maybe I misunderstand the issue, how would `fileName.Replace("\\'", "'")` help this case?

**Comment by QilongTang** [comment]
@saintentropy or @mrahmaniasl Can you also make it clear if this needs to go into Revit 2021.1 or can be included in the later point release?

File: test/DynamoCoreTests/DSLibraryTest.cs (hunk 2 above)
Comment (saintentropy) [reply test/DynamoCoreTests/DSLibraryTest.cs:143]: Sorry now I get your question @QilongTang.  The path here gets modified here

https://github.com/DynamoDS/Dynamo/blob/master/src/Engine/ProtoCore/Utils/CompilerUtils.cs#L258 in `TryLoadAssemblyIntoCore`

The return value of `ToLiteral` is modified from _Pack'age_ to _Pack\\'age_ when it is parsed here https://github.com/DynamoDS/Dynamo/blob/master/src/Engine/ProtoCore/Utils/CompilerUtils.cs#L242

Comment (saintentropy) [reply test/DynamoCoreTests/DSLibraryTest.cs:143]: image

Comment (saintentropy) [reply test/DynamoCoreTests/DSLibraryTest.cs:143]: Above is one example of the of the call to `ToLiteral`

**Comment by saintentropy** [comment]
@QilongTang Up to you regarding release.  This was originally filed by localization but is not a blocker for GD

**Comment by QilongTang** [comment]
LGTM

**Pull request merged**